.github/workflows/node.js.yml

```diff
@@ -10,31 +7,48 @@ on:
   pull_request:
 
 jobs:
-  build:
+  test:
+    name: Run tests on supported Node.js versions
     runs-on: ubuntu-latest
 
     strategy:
       fail-fast: false
       matrix:
-        node-version: [14.x, 16.x, 17.x, 18.x]
-        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+        # See supported Node.js versions at https://nodejs.org/en/about/previous-releases
+        node-version: [18, 20, 21]
 
     steps:
-    - uses: actions/checkout@main
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@main
-      with:
-        node-version: ${{ matrix.node-version }}
-    - run: npm ci
-    - run: npm test
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run tests
+        run: npm test
 
   dtslint:
+    name: Validate type definitions
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-node@v2
-      with:
-        node-version: lts/*
-    - run: npm ci
-    - run: npm run dtslint
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
```

Comment:
All Node.js workflows that are not part of the matrix are hard-coded to the latest actively supported Node.js version to prevent accidental regressions.